lib/vssh/libssh2.c

```diff
@@ -1766,8 +1766,15 @@ static CURLcode ssh_statemach_act(struct Curl_easy *data, bool *block)
           sshc->actualcode = CURLE_QUOTE_ERROR;
           break;
         }
-        sshp->quote_attrs.atime = (unsigned long)date;
-        sshp->quote_attrs.flags = LIBSSH2_SFTP_ATTR_ACMODTIME;
+#if SIZEOF_TIME_T > SIZEOF_LONG
+        if(date > 0xffffffff)
```

Comment:
I personally would use `UINT_MAX` or at least add a comment explaining that this case is for >32 bit.

Reply:
It felt wrong to use an int related define when none of the variables involved are ints and I want to make the check exactly 32 bit.

I figure we could do infof() there, but probably it should even rather return error?

Reply:
> It felt wrong to use an int related define when none of the variables involved are ints and I want to make the check exactly 32 bit.

I totally follow you here, makes sense.

> I figure we could do infof() there, but probably it should even rather return error?

I think an error would probably be even better, because some software heavily relies on the mtime of files. For example, I've seen many website generators (no matter if dynamic or static) which make use of that value in order to sort the entries. A wrong mtime would be a disaster in such cases.

Reply:
I made #9328